src/infra/accounts/postgres/src/repos/postgres_device_code_repository.rs

```diff
@@ -0,0 +1,156 @@
+            device_code_created.expires_at,
+        )
+        .execute(connection_mut)
+        .await
+        .int_err()?;
+
+        Ok(())
+    }
+
+    async fn update_device_token_with_token_params_part(
+        &self,
+        device_code: &DeviceCode,
+        token_params_part: &DeviceTokenParamsPart,
+    ) -> Result<(), UpdateDeviceCodeWithTokenParamsPartError> {
+        let mut tr = self.transaction.lock().await;
+
+        let connection_mut = tr.connection_mut().await?;
+
+        let account_id = token_params_part.account_id.as_did_str().to_stack_string();
+        let token_iat: i64 = token_params_part.iat.try_into().int_err()?;
+        let token_exp: i64 = token_params_part.exp.try_into().int_err()?;
+
+        let update_result = sqlx::query!(
+            r#"
+            UPDATE device_codes
+            SET token_iat  = $2,
+                token_exp  = $3,
+                account_id = $4
+            WHERE device_code = $1
+            "#,
+            device_code.as_ref(),
+            token_iat,
+            token_exp,
+            account_id.as_str(),
+        )
+        .execute(connection_mut)
+        .await
+        .int_err()?;
+
+        if update_result.rows_affected() == 0 {
+            return Err(DeviceTokenFoundError {
+                device_code: device_code.clone(),
+            }
+            .into());
+        }
+
+        Ok(())
+    }
+
+    async fn find_device_token_by_device_code(
+        &self,
+        device_code: &DeviceCode,
+    ) -> Result<DeviceToken, FindDeviceTokenByDeviceCodeError> {
+        let mut tr = self.transaction.lock().await;
+
+        let connection_mut = tr.connection_mut().await?;
+
+        let maybe_device_token_row_model = sqlx::query_as!(
+            DeviceTokenRowModel,
+            r#"
+            SELECT device_code,
+                   device_code_created_at,
+                   device_code_expires_at,
+                   token_iat,
+                   token_exp,
+                   token_last_used_at,
+                   account_id
+            FROM device_codes
+            WHERE device_code = $1
+            "#,
+            device_code.as_ref()
+        )
+        .fetch_optional(connection_mut)
+        .await
+        .int_err()?;
+
+        if let Some(device_token_row_model) = maybe_device_token_row_model {
+            let device_token = device_token_row_model.try_into()?;
+
+            Ok(device_token)
+        } else {
+            Err(DeviceTokenFoundError {
+                device_code: device_code.clone(),
+            }
+            .into())
+        }
+    }
+
+    async fn cleanup_expired_device_codes(&self) -> Result<(), CleanupExpiredDeviceCodesError> {
+        let mut tr = self.transaction.lock().await;
+
+        let connection_mut = tr.connection_mut().await?;
+
+        sqlx::query!(
+            r#"
+            DELETE
+            FROM device_codes
+            WHERE device_code_expires_at < NOW()
```

Comment:
Is this taking timezone differences into account?

Reply:
For record: after writing the tests, no additional changes were required